### PR TITLE
Improvement/file test coverage

### PR DESCRIPTION
**Pull request by justinehell**
Add coverage for `ModelizerDrawPage.vue` file

### CODE REVIEW

File: tests/unit/pages/ModelizerDrawPage.spec.js [hunk 1]
```diff
@@ -39,14 +46,79 @@ jest.mock('src/composables/PluginManager', () => ({
...
   });
 
+  describe('Test function: onDefaultEvent', () => {
+    it('should call renderConfiguration if event.action is "move"', async () => {
+      const param = {
+        event: {
+          type: 'Drawer', action: 'move',
+        },
+      };
+      await wrapper.vm.onDefaultEvent(param);
```
Comment (k-y0u) [inline tests/unit/pages/ModelizerDrawPage.spec.js:87]: Jump

File: tests/unit/pages/ModelizerDrawPage.spec.js [hunk 2]
```diff
@@ -39,14 +46,79 @@ jest.mock('src/composables/PluginManager', () => ({
...
+  let unsubscribe;
 
   beforeEach(() => {
+    subscribe = jest.fn();
+    unsubscribe = jest.fn();
+
+    PluginEvent.DefaultEvent.subscribe.mockImplementation(() => {
+      subscribe();
+      return { unsubscribe };
+    });
```
Comment (k-y0u) [inline tests/unit/pages/ModelizerDrawPage.spec.js:76]: Jump after

File: tests/unit/pages/ModelizerDrawPage.spec.js [hunk 3]
```diff
@@ -39,14 +46,79 @@ jest.mock('src/composables/PluginManager', () => ({
...
+      expect(PluginManager.renderConfiguration).toBeCalledWith('project-00000000', 'path', wrapper.vm.data.plugin);
+    });
+
+    it('should call renderModel if event.action is either "update", "delete", or "add"', async () => {
+      let param = {
+        event: {
+          type: 'Drawer', action: 'update',
+        },
+      };
+      await wrapper.vm.onDefaultEvent(param);
```
Comment (k-y0u) [inline tests/unit/pages/ModelizerDrawPage.spec.js:98]: jump before

File: tests/unit/pages/ModelizerDrawPage.spec.js [hunk 4]
```diff
@@ -39,14 +46,79 @@ jest.mock('src/composables/PluginManager', () => ({
...
+        event: {
+          type: 'Drawer', action: 'move',
+        },
+      };
+      await wrapper.vm.onDefaultEvent(param);
+
+      expect(PluginManager.renderConfiguration).toBeCalledWith('project-00000000', 'path', wrapper.vm.data.plugin);
+    });
+
+    it('should call renderModel if event.action is either "update", "delete", or "add"', async () => {
```
Comment (k-y0u) [inline tests/unit/pages/ModelizerDrawPage.spec.js:92]: remove coma after "delete"

File: tests/unit/pages/ModelizerDrawPage.spec.js [hunk 5]
```diff
@@ -55,4 +127,81 @@ describe('Test page component: ModelizerDrawPage', () => {
...
+  });
+
+  describe('Test function: dropHandler', () => {
+    it('should call addNewComponent if component is not a template', async () => {
+      const param = {
+        dataTransfer: {
+          getData: jest.fn(() => ('{"isTemplate":false,"definitionType":"testComponent"}')),
+        },
+      };
+      await wrapper.vm.dropHandler(param);
```
Comment (k-y0u) [inline tests/unit/pages/ModelizerDrawPage.spec.js:161]: jump

File: tests/unit/pages/ModelizerDrawPage.spec.js [hunk 6]
```diff
@@ -55,4 +127,81 @@ describe('Test page component: ModelizerDrawPage', () => {
...
+      expect(PluginManager.renderModel).toBeCalledWith('project-00000000', 'path', wrapper.vm.data.plugin);
+    });
+
+    it('should call addNewTemplateComponent if component is a template', async () => {
+      const param = {
+        dataTransfer: {
+          getData: jest.fn(() => ('{"isTemplate":true,"definitionType":"testComponent"}')),
+        },
+      };
+      await wrapper.vm.dropHandler(param);
```
Comment (k-y0u) [inline tests/unit/pages/ModelizerDrawPage.spec.js:173]: jump

File: tests/unit/pages/ModelizerDrawPage.spec.js [hunk 7]
```diff
@@ -55,4 +127,81 @@ describe('Test page component: ModelizerDrawPage', () => {
...
+    it('should emit a notification on error when adding a new template component', async () => {
+      Notify.create = jest.fn();
+      PluginManager.addNewTemplateComponent.mockImplementation(() => Promise.reject());
+
+      const param = {
+        dataTransfer: {
+          getData: jest.fn(() => ('{"isTemplate":true,"definitionType":"testComponent"}')),
+        },
+      };
+      await wrapper.vm.dropHandler(param);
```
Comment (k-y0u) [inline tests/unit/pages/ModelizerDrawPage.spec.js:188]: jump

**Review by k-y0u — CHANGES_REQUESTED** [review]

File: tests/unit/pages/ModelizerDrawPage.spec.js (hunk 1 above)
Comment (justinehell) [reply tests/unit/pages/ModelizerDrawPage.spec.js:87]: done

File: tests/unit/pages/ModelizerDrawPage.spec.js (hunk 2 above)
Comment (justinehell) [reply tests/unit/pages/ModelizerDrawPage.spec.js:76]: done

File: tests/unit/pages/ModelizerDrawPage.spec.js (hunk 4 above)
Comment (justinehell) [reply tests/unit/pages/ModelizerDrawPage.spec.js:92]: done

File: tests/unit/pages/ModelizerDrawPage.spec.js (hunk 5 above)
Comment (justinehell) [reply tests/unit/pages/ModelizerDrawPage.spec.js:161]: done

File: tests/unit/pages/ModelizerDrawPage.spec.js (hunk 6 above)
Comment (justinehell) [reply tests/unit/pages/ModelizerDrawPage.spec.js:173]: done

**Review by k-y0u — APPROVED** [review]

**Review by aboubakar14 — APPROVED** [review]

**Pull request merged**